nestjs/src/cross-check/cross-check.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Repository } from 'typeorm';
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { HttpService } from '@nestjs/axios';
+import { Cron } from '@nestjs/schedule';
+import { ConfigService } from '../config';
+import { getTaskFilters } from './utils/tasksFiltering';
+import { DEV_HOST, PROD_HOST } from './utils/constants';
+import { CourseTask } from '@entities/courseTask';
+import { from, catchError, mergeMap, EMPTY, toArray, lastValueFrom } from 'rxjs';
+
+@Injectable()
+export class CrossCheckService {
+  private readonly logger = new Logger('CrossCheckService');
+
+  constructor(
+    @InjectRepository(CourseTask)
+    private readonly courseTaskRepository: Repository<CourseTask>,
+    private readonly httpService: HttpService,
+    private readonly conf: ConfigService,
+  ) {}
+
+  @Cron('5 0 * * *', { timeZone: 'UTC' })
```

Comment:
wondering is needed to call this job so often?

Reply:
This means means "every day once a day at 00:05 UTC". This is the lowest possible frequency for this operation, as we have to update the CourseTask status daily.
![image](https://user-images.githubusercontent.com/34455330/160241525-2db5b53b-c8c7-42b9-b914-f4a44d85a8be.png)

Reply:
I am didn't check, but according to documentation https://docs.nestjs.com/techniques/task-scheduling first value is second

Reply:
It's optional :)
If you don't provide seconds, it starts with minutes

Reply:
understood, didn't pay attention to number of symbols)